src/Math-Numerical/Integer.extension.st

```diff
@@ -1,5 +1,54 @@
 Extension { #name : #Integer }
 
+{ #category : #'*Math-Numerical' }
+Integer >> findNK [
+	" Reverse binomial coefficient. Answer a <Collection> with all n and k such that n take: k = self. Elements in the answered Collection should be read as paired. Each pair represents (n,k) in the binomial coefficient formula. 
+	See https://math.stackexchange.com/a/103385/205 for details. "
+
+	| k |
+	[ self isInteger and: [ self > 1 ] ] assert.
+    k := 0.
+	^ Array streamContents: [ : stream | 
+	    [ true ] whileTrue: [ 
+			| nmin nmax choose |
+			k := k + 1.
+			((2 * k + 1) * self) <= (4 ** k)
+				ifTrue: [ ^ stream contents ].
+			nmin := k firstOver: (k factorial * self).
+			nmax := nmin + k + 1.
+			nmin := nmin max: (2 * k).
+	      choose := nmin asInteger take: k.
+			nmin to: nmax do: [ : n |
+				choose = self
+					ifTrue: [  
+						stream nextPutAll: { n asInteger . k asInteger }.
+						k < (n - k)
+							ifTrue: [ stream nextPutAll: { n asInteger . (n - k) asInteger } ] ].
+	            choose := choose * (n + 1).
+	            choose := (choose / (n + 1 - k)) ceiling ] ] ]
+
+]
+
+{ #category : #'*Math-Numerical' }
+Integer >> firstOver: c [
```

Comment:
Maybe add a link to the underlying Mathematics? This is a part of the domain I'm not clear on.